File: implementacao c++/Genero.cpp
#include "Genero.hpp"



int getIdGenero(Genero genero) {
    
    return static_cast<int>(genero);
}

Genero generoCandidato(int codigo){
    switch (codigo) {
        case 2:
            return Genero::MASCULINO;
        case 4:
            return Genero::FEMININO;
        default:
            throw std::invalid_argument("Código de gênero inválido: " + std::to_string(codigo));
    }
}

File: implementacao java/src/Dominio/Genero.java
package Dominio;
public enum Genero{
    
    MASCULINO(2),FEMININO(4);


    private int id;

    Genero(int id){
        this.id = id;
    }

    public int getIdGenero(){
        return id;
    }

    public static Genero generoCandidato(int codigo){
        
        for(Genero genero: values()){
            if(genero.id == codigo){
                return genero;
            }
        }
        throw new IllegalArgumentException("Código de gênero inválido: " + codigo);
    }
}